donut/modules/bodfeedback/helpers.py

```diff
@@ -0,0 +1,255 @@
+    '''
+    query = 'SELECT uuid FROM bod_complaint_info WHERE complaint_id = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchone()
+        if not res:
+            return None
+        uuid = res['uuid']
+    return flask.url_for(
+        'bodfeedback.bodfeedback_view_complaint', id=uuid, _external=True)
+
+
+def get_id(uuid):
+    '''
+    Returns the complaint_id associated with a uuid
+    or false if the uuid is not found
+    '''
+    query = 'SELECT complaint_id FROM bod_complaint_info WHERE uuid = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, str(uuid))
+        if not cursor.rowcount:
+            return False
+    return cursor.fetchone()['complaint_id']
+
+
+def get_messages(complaint_id):
+    '''
+    Returns timestamps, posters, messages, and message_id's on this complaint
+    in ascending order of timestamp
+    '''
+    query = """
+    SELECT time, poster, message, message_id FROM bod_complaint_messages WHERE complaint_id = %s ORDER BY time
+    """
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchall()
+    if not (res and 'message_id' in res[0]):
+        return None
+    return res
+
+
+def get_summary(complaint_id):
+    '''
+    Returns a dict with the following fields: subject, status
+    '''
+    query = 'SELECT subject, status FROM bod_complaint_info WHERE complaint_id = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchone()
+    return res
+
+
+def get_subject(complaint_id):
+    '''
+    Returns the suject or None if complaint_id is invalid
+    '''
+    res = get_summary(complaint_id)
+    return res['subject'] if res else None
+
+
+def get_status(complaint_id):
+    '''
+    Returns the status of a post or None if complaint_id is invalid
+    '''
+    res = get_summary(complaint_id)
+    return res['status'] if res else None
+
+
+def mark_read(complaint_id):
+    '''
+    Sets the status of this complaint to 'read'
+    returns False if complaint_id is invalid
+    '''
+    query = """
+    UPDATE bod_complaint_info SET status = 'read' WHERE complaint_id = %s
+    """
+    if get_status(complaint_id) is None:
+        return False
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+
+
+def mark_unread(complaint_id):
+    '''
+    Sets the status of this complaint to 'new_msg'
+    returns False if complaint_id is invalid
+    '''
+    query = """
+    UPDATE bod_complaint_info SET status = 'new_msg' WHERE complaint_id = %s
+    """
+    if get_status(complaint_id) is None:
+        return False
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        # Not sure why this doesn't autocommit
+        cursor.execute("COMMIT;")
```

Comment:
this could be why [https://stackoverflow.com/a/35621700](link). @calebsander do you think we should enable autocommit? or otherwise this is probably fine.

Reply:
We already have `autocommit` turned on: https://github.com/ASCIT/donut/blob/865dccd4c190078a7d856cb13227007d014808aa/donut/__init__.py#L85